common/set_test.cpp

```diff
@@ -58,10 +59,19 @@ auto MakeElements(RangeT&& range, RangeTs&&... ranges) {
 template <typename SetT>
 class SetTest : public ::testing::Test {};
 
-using Types = ::testing::Types<Set<int>, Set<int, 16>, Set<int, 128>,
-                               Set<TestData>, Set<TestData, 16>>;
+template <typename SetT>
+class CopyableSetTest : public ::testing::Test {};
+
+using Types =
+    ::testing::Types<Set<int>, Set<int, 16>, Set<int, 128>, Set<TestData>,
+                     Set<TestData, 16>, Set<MoveOnlyTestData>,
+                     Set<MoveOnlyTestData, 16>>;
 TYPED_TEST_SUITE(SetTest, Types);
```

Comment:
What do you think about having a small / narrow test for move-only types and the restricted map API? I feel like that would let us avoid having the complexity of handling move-only types and skipping some of the tests in the suite below?

Reply:
Sure, I think that would be okay, done.